smack-java7/src/main/java/org/jivesoftware/smack/java7/XmppHostnameVerifier.java

```diff
@@ -133,7 +136,20 @@ private static void matchDns(String name, X509Certificate cert) throws Certifica
                 throw new CertificateException(sb.toString());
             }
         }
-        // TODO SubjectX500Name
+        try {
+            LdapName dn = new LdapName(cert.getSubjectX500Principal().getName());
+            for (Rdn rdn : dn.getRdns()) {
+                if (rdn.getType().equalsIgnoreCase("CN")) {
+                    if (matchesPerRfc2818(name, rdn.getValue().toString())) {
+                        return;
+                    }
+                    break;
+                }
+            }
+        } catch (Exception e) {
```

Comment:
Ahh, I see. Please make the catch more specific and restrict the try block to only the places where the throw is expected to occur. I'm also not sure if the log level should be warning, all other logs are just 'info'. But maybe a warning is justified in this case.